script/update-archive-version.ts

```diff
@@ -35,22 +59,26 @@ async function updateArchiveVersion() {
   );
 
   const newBookLoader = makeUnifiedBookLoader(
-    createArchiveLoader(args.newArchive, {
+    createArchiveLoader(`${REACT_APP_ARCHIVE_URL_BASE}${args.pipelineVersion}`, {
       archivePrefix: ARCHIVE_URL,
     }),
     osWebLoader
   );
 
   const updateRedirectsPromises: Array<() => Promise<[BookWithOSWebData, number]>> = [];
-  const bookEntries = Object.entries(BOOKS);
-
-  console.log('Preparing books...');
+  for (const book of booksToUpdate) {
+    await processBookVersionUpdate(book, `${REACT_APP_ARCHIVE_URL_BASE}${args.pipelineVersion}`);
+  }
+  const bookEntries = Object.entries(BOOKS_CONFIG);
 
+  console.log('Preparing redirects...');
   for (const [bookId, { defaultVersion }] of bookEntries) {
```

Comment:
shouldn't it skip everything not specified in `booksToUpdate` if there is no updated pipeline?

Reply:
this doesn't run if there's no updated pipeline (see lines 39–47)